File: sem2/ADF-I/java-1/d06_abstract/src/data/Shape.java
/*
 * Day la lop truu tuong mo ta cau truc 1 doi tuong hinh hoc tong quat
 * co 2 ham tinh dien tich va tinh chu vi
 */
package data;

public abstract class Shape {

    public abstract double getArea();

    public abstract double getPeripheral();
    
    //ham truu tuong nhap du lieu cho 1 doi tuong hinh hoc bat ky
    public abstract void input();

    //ham in ra dien tich va chu cua 1 doi tuong hinh hoc
    public void output() {
        System.out.printf("Dien tich: %.2f \n", getArea());
        System.out.printf("Chu vi: %.2f \n", getPeripheral());
    }
}

File: sem2/ap-assignments/AP/d03_abstract/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace d03_abstract
{
    public abstract class Shape
    {
        //phuong thuc truu tuong tinh dien tich 1 doi tuong hinh hoc
        public abstract float Area();      //khi la abstract thì ko cần virtual
        //phuong thuc truu tuong tinh chu vi 1 dt hinh hoc
        public abstract float Peripheral();
        // Hang so PI
        public const float PI = 3.14f;

        public override string ToString()
        {
            return string.Format($"Dien Tich:{Area()},Chu Vi:{Peripheral()}");
        }
    }
}
